SSL_OP_ALL will be zero in LibreSSL >= 4.4

A [recent commit][1] removed `SSL_OP_LEGACY_SERVER_CONNECT` from LibreSSL's `SSL_OP_ALL`, which is now 0 like in BoringSSL and aws-lc (this will be the case starting from LibreSSL 4.4.0).

This results in two assertion failures in `test_ssl_ctx_options` and `test_ssl_options` (tested with cryptography 47.0.0 and 49.0.0). Not sure what the preferred fix for this would be. Just skip the two assertions for `CRYPTOGRAPHY_IS_LIBRESSL` or is it worth having a version dependency here?

```
=================================== FAILURES ===================================
_______________________ TestOpenSSL.test_ssl_ctx_options _______________________

self = <test_openssl.TestOpenSSL object at 0xcec16b09d10>

    def test_ssl_ctx_options(self):
        # Test that we're properly handling 32-bit unsigned on all platforms.
        b = Binding()
        # SSL_OP_ALL is 0 on BoringSSL
        if not (
            rust_openssl.CRYPTOGRAPHY_IS_BORINGSSL
            or rust_openssl.CRYPTOGRAPHY_IS_AWSLC
        ):
>           assert b.lib.SSL_OP_ALL > 0
E           AssertionError: assert 0 > 0
E            +  where 0 = <module 'lib'>.SSL_OP_ALL
E            +    where <module 'lib'> = <cryptography.hazmat.bindings.openssl.binding.Binding object at 0xcec3d323ed0>.lib

tests/hazmat/bindings/test_openssl.py:31: AssertionError
_________________________ TestOpenSSL.test_ssl_options _________________________

self = <test_openssl.TestOpenSSL object at 0xcebd8343e10>

    def test_ssl_options(self):
        # Test that we're properly handling 32-bit unsigned on all platforms.
        b = Binding()
        # SSL_OP_ALL is 0 on BoringSSL
        if not (
            rust_openssl.CRYPTOGRAPHY_IS_BORINGSSL
            or rust_openssl.CRYPTOGRAPHY_IS_AWSLC
        ):
>           assert b.lib.SSL_OP_ALL > 0
E           AssertionError: assert 0 > 0
E            +  where 0 = <module 'lib'>.SSL_OP_ALL
E            +    where <module 'lib'> = <cryptography.hazmat.bindings.openssl.binding.Binding object at 0xcec35b41e00>.lib

tests/hazmat/bindings/test_openssl.py:49: AssertionError
```

Cc @sthen

[1]: https://github.com/openbsd/src/commit/969f618254fae3c010e41876e452966bc7878117

--- a/tests/hazmat/bindings/test_openssl.py
+++ b/tests/hazmat/bindings/test_openssl.py
@@ -23,10 +23,11 @@ class TestOpenSSL:
     def test_ssl_ctx_options(self):
         # Test that we're properly handling 32-bit unsigned on all platforms.
         b = Binding()
-        # SSL_OP_ALL is 0 on BoringSSL
+        # SSL_OP_ALL is 0 on BoringSSL, AWS-LC and recent LibreSSL
         if not (
             rust_openssl.CRYPTOGRAPHY_IS_BORINGSSL
             or rust_openssl.CRYPTOGRAPHY_IS_AWSLC
+            or rust_openssl.CRYPTOGRAPHY_IS_LIBRESSL
         ):
             assert b.lib.SSL_OP_ALL > 0
         ctx = b.lib.SSL_CTX_new(b.lib.TLS_method())
@@ -41,10 +42,11 @@ class TestOpenSSL:
     def test_ssl_options(self):
         # Test that we're properly handling 32-bit unsigned on all platforms.
         b = Binding()
-        # SSL_OP_ALL is 0 on BoringSSL
+        # SSL_OP_ALL is 0 on BoringSSL, AWS-LC and recent LibreSSL
         if not (
             rust_openssl.CRYPTOGRAPHY_IS_BORINGSSL
             or rust_openssl.CRYPTOGRAPHY_IS_AWSLC
+            or rust_openssl.CRYPTOGRAPHY_IS_LIBRESSL
         ):
             assert b.lib.SSL_OP_ALL > 0
         ctx = b.lib.SSL_CTX_new(b.lib.TLS_method())

